akka-actor/src/main/scala/akka/actor/LightArrayRevolverScheduler.scala

```diff
@@ -54,13 +54,15 @@ class LightArrayRevolverScheduler(
 
   import LightArrayRevolverScheduler._
 
-  private val oneNs = Duration.fromNanos(1l)
-  private def roundUp(d: FiniteDuration): FiniteDuration =
-    try {
-      ((d + TickDuration - oneNs) / TickDuration).toLong * TickDuration
-    } catch {
-      case _: IllegalArgumentException ⇒ d // rounding up Long.MaxValue.nanos overflows
+  private def roundUp(d: FiniteDuration): FiniteDuration = {
+    val dn = d.toNanos
+    if (dn <= 0)
+      Duration.Zero
+    else {
+      val r = ((dn - 1) / tickNanos + 1) * tickNanos
+      if (r != dn && r > 0) r.nanos else d
```

Comment:
Not sure the extra work for `dn <= 0` really matters, as:

```scala
if (delay <= Duration.Zero) {
      if (stopped.get != null) throw new SchedulerException("cannot enqueue after timer shutdown")
      ec.execute(r)
      NotCancellable
```

later in the code where the rounded up value is passed to. So anything <= Zero is executed immediately anyway.

So it's more about the edge case of dn == MinValue where it would wrap around in the else... Not sure we really care about that though.